models/optional_modules/core_web_vitals/snowflake/snowplow_web_vital_measurements.sql

```diff
@@ -0,0 +1,262 @@
+
+  select distinct
+
+    url_group as page_url,
+    device_class,
+    'all' as country,
+    'last {{var('snowplow__cwv_days_to_measure')|string }} days' as time_period,
+    count(*) over (partition by url_group) as page_view_count,
+    nullif(percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by lcp) over (partition by url_group), 0) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by fid) over (partition by url_group) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by cls) over (partition by url_group) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by ttfb) over (partition by url_group) as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+    'by_url_group' as measurement_type
+
+  from {{ ref('snowplow_web_vitals') }}
+
+  where derived_tstamp >= {{ snowplow_utils.timestamp_add('day', '-'+var('snowplow__cwv_days_to_measure')|string, snowplow_utils.current_timestamp_in_utc()) }}
+
+)
+
+, measurements as (
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_url_and_device
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from overall
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_device
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_day
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_day_and_device
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_country
+
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_country_and_device
+
+  union all
+
+  select *,
+
+  {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from by_url_group
+
+)
+
+select
+  measurement_type,
+  page_url,
+  device_class,
+  country,
+  time_period,
+  page_view_count,
+  ceil(lcp_{{ var('snowplow__cwv_percentile') }}p *1000) / 1000 as lcp_{{ var('snowplow__cwv_percentile') }}p,
+  ceil(fid_{{ var('snowplow__cwv_percentile') }}p *1000) / 1000 as fid_{{ var('snowplow__cwv_percentile') }}p,
+  ceil(cls_{{ var('snowplow__cwv_percentile') }}p *1000) / 1000 as cls_{{ var('snowplow__cwv_percentile') }}p,
+  ceil(ttfb_{{ var('snowplow__cwv_percentile') }}p *1000) / 1000 as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+  lcp_result,
```

Comment:
Same here about the ceil precision thing, if that is what this is trying to do.

Reply:
it is what I'm trying to do but it doesn't seem to exist for bigquery

Reply:
BQ going down in my tier list, this is the snowflake file though